batch10/assignment3/HASSAN_SHAIKH.md

```diff
@@ -0,0 +1,26 @@
+Q1) Complete all the labs - Done
+
+Q2) How to leverage cache using Dockerfiles 
+>> Dockerfile help to avoid downloading image again and again.
+
+Q3) What are multi-stage builds 
+>> You can use different container to build a binary and than release that build by simply using the binary in a seprate image whcih does not contain the run time or build time of the language. 
+
+Q4) Explain containers vs Image 
+>> Images are the pre build package where as container is a run time of that image. 
+
+Q5) Explain RUN vs CMD vs Entrypoint 
+>> First thing that execute is Entrypoint, followied by CMD. RUN is used to execute any command. 
```

Comment:
Wrong, CMD is passed as argument to Entrypoint...
RUN is build time command used to execute a command and make it part of image